<launch>

    <!-- This file contains the launch facility for the hardware interface.
    Therefore, one should include this file in all the controller launch files. -->

    <arg name="data_subfolder"/>
    <arg name="sim" default="false"/>
    <arg name="rtprio_nodes" default="false"/>
    <arg name="hardware_connected" default="false"/>
    <arg name="run_rviz" default="true"/>
    <arg name="run_rqt_multiplot" default="false"/>
    <arg name="record_data" default="true"/>

    <!-- Parameters -->
    <rosparam command="load" file="$(find oct_levitation)/config/general_params.yaml"/>
    <rosparam command="load" file="$(find oct_levitation)/config/controller_params.yaml"/>

    <group if="$(arg record_data)">
        <node pkg="oct_levitation" name="experiment_data_recorder" type="experiment_recorder.py" output="screen">
            <param name="sim" value="$(arg sim)"/>
            <param name="data_subfolder" value="$(arg data_subfolder)"/>
        </node>
    </group>

    <group if="$(hardware_connected)">
        <include file="$(find oct_levitation)/launch/tnb_mns_driver.launch"/>
    </group>

    <!-- This node can be run without rviz, it publishes some important details. -->
    <node pkg="oct_levitation" name="rviz_visualizations_node" type="rviz_visualizations.py" output="screen"/>
    <group if="$(arg run_rviz)">
        <node pkg="rviz" type="rviz" name="rviz" args="-d $(find oct_levitation)/config/rviz/single_dipole_control_config.rviz" output="screen"/>
    </group>

    <group if="$(arg run_rqt_multiplot)">
        <node pkg="rqt_multiplot" type="rqt_multiplot" name="rqt_multiplot" output="screen"
            args="--force-config $(find your_package)/config/rqt_multiplot/rqt_multiplot_pose_config.xml" />
    </group>
</launch>